### ISSUE TITLE
SDL2_mixer build fails (0.5.3)

### ISSUE THREAD

**Issue by yegortimoshenko** [EasyRPG/Player#1309]
Build with SDL2_mixer fails (on Linux amd64). Feel free to close if this is a duplicate. Logs:

```
cmake flags: -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/44ahrwm4w8k0hl2vhka91aplv3a9ybx3-easyrpg-player-0.5.3/include -DCMAKE_INSTALL_LIBDIR=/nix/store/44ahrwm4w8k0hl2vhka91aplv3a9ybx3-easyrpg-player-0.5.3/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/44ahrwm4w8k0hl2vhka91aplv3a9ybx3-easyrpg-player-0.5.3/lib -DCMAKE_INSTALL_PREFIX=/nix/store/44ahrwm4w8k0hl2vhka91aplv3a9ybx3-easyrpg-player-0.5.3  
-- The CXX compiler identification is GNU 6.4.0
-- The C compiler identification is GNU 6.4.0
-- Check for working CXX compiler: /nix/store/yjh21h38fmw3i4h2b5r557p4ggbvrd0v-gcc-wrapper-6.4.0/bin/g++
-- Check for working CXX compiler: /nix/store/yjh21h38fmw3i4h2b5r557p4ggbvrd0v-gcc-wrapper-6.4.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /nix/store/yjh21h38fmw3i4h2b5r557p4ggbvrd0v-gcc-wrapper-6.4.0/bin/gcc
-- Check for working C compiler: /nix/store/yjh21h38fmw3i4h2b5r557p4ggbvrd0v-gcc-wrapper-6.4.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Found liblcf: /nix/store/48r9qazyvr6gzg87hbk84dmddavnvpz0-liblcf-0.5.3/include/liblcf  
-- Found Pixman: /nix/store/swi9n1azsvv36bsbpm98pn87zlsm6vi6-pixman-0.34.0/include/pixman-1  
-- Found ZLIB: /nix/store/qz2lprii3jxy0wybggj4h3304ij7m56p-zlib-1.2.11/lib/libz.so (found version "1.2.11") 
-- Found PNG: /nix/store/a83mwdh3xla231kjk1w475awzyk1yhfq-libpng-apng-1.6.34/lib/libpng.so (found version "1.6.34") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found SDL2: /nix/store/l084zhickcwv71ygkxywwsg5x4769yi6-SDL2-2.0.7-dev/lib/libSDL2main.a;/nix/store/7q676sg2v91za8qkc10290xg3krxi1d4-SDL2-2.0.7/lib/libSDL2.so;-lpthread (found version "2.0.7") 
-- Found Freetype: /nix/store/6b12yg3y7dbak7k99fxkz3n0fzaz0r0g-freetype-2.7.1/lib/libfreetype.so (found version "2.7.1") 
-- Found Harfbuzz: /nix/store/fy80flz98lzidwyk6ls866f621qbaq46-harfbuzz-1.5.1-dev/include/harfbuzz  
-- Found SDL2_mixer: /nix/store/9a8ylga3lbxy7zxdm7jravf250a1pgwa-SDL2_mixer-2.0.2/lib/libSDL2_mixer.so (found version "2.0.2") 
-- Found speexdsp: /nix/store/jalldr4dl0qv0f2m0dim2k10yvf9dgav-speexdsp-1.2rc3-dev/include  
-- Found mpg123: /nix/store/94qpxjphw4w0gfl6288g0z6j95h8m5dj-mpg123-1.25.7/include  
-- Found SndFile: /nix/store/cmk2jvb4z0zmz8paw7lavi5sbplanhrw-libsndfile-1.0.28-dev/include  
-- Found OggVorbis: /nix/store/gra9xzanz75zyx244ql43ny4c5c55ch5-libvorbis-1.3.5-dev/include  
-- Found Opus: /nix/store/07368mpmxa96zp630aayi6w4rxh1icln-opusfile-0.8/include/opus  
-- Found XMP: /nix/store/0ilyj779gfka19b8v5nvzrndqb5p8xqh-libxmp-4.3.12/include  
-- Found Doxygen: /nix/store/s4w8ypg06y0y1p8pvp66xhig2jgd8c43-doxygen-1.8.11/bin/doxygen (found version "1.8.11") found components:  doxygen missing components:  dot
-- 
-- Audio backend: SDL2_mixer
-- 
-- WAV playback:  libsndfile
-- MIDI playback: SDL2_mixer. Fallback to built-in (FmMidi)
-- MP3 playback:  mpg123
-- Ogg Vorbis playback: libvorbis
-- MOD playback:  libxmp
-- Opus playback: opusfile
-- Resampler:     speexdsp
-- 
-- Font rendering: Freetype with Harfbuzz / built-in (Shinonome)
-- 
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR


-- Build files have been written to: /tmp/nix-build-easyrpg-player-0.5.3.drv-0/source/build
building
build flags: SHELL=/nix/store/4ada72n7785wwazv42fhsnxjvilaa3aj-bash-4.4-p12/bin/bash
Scanning dependencies of target EasyRPG_Player_Static
[  0%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/async_handler.cpp.o

[  1%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio_al.cpp.o
[  1%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio.cpp.o
[  2%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio_decoder.cpp.o
[  3%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio_generic.cpp.o
[  3%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio_resampler.cpp.o
[  4%] Building CXX object CMakeFiles/EasyRPG_Player_Static.dir/src/audio_sdl_mixer.cpp.o
/tmp/nix-build-easyrpg-player-0.5.3.drv-0/source/src/audio_sdl_mixer.cpp: In member function 'void SdlMixerAudio::BGM_OnPlayedOnce()':
/tmp/nix-build-easyrpg-player-0.5.3.drv-0/source/src/audio_sdl_mixer.cpp:246:38: error: 'MUS_MP3_MAD' was not declared in this scope
   if (Mix_GetMusicType(bgm.get()) != MUS_MP3_MAD)
                                      ^~~~~~~~~~~
make[2]: *** [CMakeFiles/EasyRPG_Player_Static.dir/build.make:207: CMakeFiles/EasyRPG_Player_Static.dir/src/audio_sdl_mixer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:211: CMakeFiles/EasyRPG_Player_Static.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

**Comment by fdelapena** [comment]
Hi, thanks for the report.

The issue appears due to a change in SDL_mixer 2.0.2 and got fixed already in master since EasyRPG/Player#1297. It is expected for the next release.
You may consider applying the patch from the related commit in the meanwhile or try the master version.

Regards.

**Comment by yegortimoshenko** [comment]
I see. Well, SDL2_mixer is an optional dependency, so I don't mind building without it.
